src/pages/home/report/ReportActionsList.js

```diff
@@ -350,7 +350,7 @@ function ReportActionsList({
             if (!currentUnreadMarker) {
                 const nextMessage = sortedReportActions[index + 1];
                 const isCurrentMessageUnread = isMessageUnread(reportAction, lastReadTimeRef.current);
-                shouldDisplay = isCurrentMessageUnread && (!nextMessage || !isMessageUnread(nextMessage, lastReadTimeRef.current));
+                shouldDisplay = isCurrentMessageUnread && (!nextMessage || !isMessageUnread(nextMessage, lastReadTimeRef.current)) && !ReportActionsUtils.shouldHideNewMarker(reportAction);
```

Comment:
Just to confirm:
This PR fully reverts https://github.com/Expensify/App/pull/32474 and this line is the real fix of https://github.com/Expensify/App/issues/31637, right?

Reply:
Yes